test/eetcd_lease_SUITE.erl

```diff
@@ -17,45 +18,60 @@ groups() ->
 
 init_per_suite(Config) ->
     application:ensure_all_started(eetcd),
-    {ok, _Pid} = eetcd:open(?Name, ["127.0.0.1:2379", "127.0.0.1:2479", "127.0.0.1:2579"]),
+    {ok, _Pid} = eetcd:open(?Name, ["127.0.0.1:2379", "127.0.0.1:2479", "127.0.0.1:2579"],
+                            [{mode, random}], tcp, []),
     Config.
 
-end_per_suite(_Config) ->
+init_per_testcase(_TestCase, Config) ->
+    revoke_all_leases(?Name),
+    Config.
+
+end_per_testcase(_TestCase, Config) ->
+    revoke_all_leases(?Name),
+    Config.
+
+end_per_suite(Config) ->
+    revoke_all_leases(?Name),
     eetcd:close(?Name),
     application:stop(eetcd),
-    ok.
+    Config.
 
 lease_base(_Config) ->
-    TTL = 100,
+    TTL = 10,
     {ok, #{'ID' := ID, 'TTL' := TTL}} = eetcd_lease:grant(?Name, TTL),
-    {ok, #{leases := [#{'ID' := ID}]}} = eetcd_lease:leases(?Name),
+    timer:sleep(200),
+    Leases1 = list_leases(?Name),
+    ?assert(lists:any(fun(#{'ID' := Val}) -> Val =:= ID end, Leases1)),
     {ok, #{}} = eetcd_lease:revoke(?Name, ID),
-    {ok, #{leases := Leases}} = eetcd_lease:leases(?Name),
-    false = lists:member(#{'ID' => ID}, Leases),
+    Leases2 = list_leases(?Name),
+    ?assertNot(lists:any(fun(#{'ID' := Val}) -> Val =:= ID end, Leases2)),
     ok.
 
 lease_timeout(_Config) ->
     TTL = 2,
     {ok, #{'ID' := ID, 'TTL' := NewTTL}} = eetcd_lease:grant(?Name, TTL),
-    {ok, #{leases := [#{'ID' := ID}]}} = eetcd_lease:leases(?Name),
+    Leases1 = list_leases(?Name),
+    ?assert(lists:any(fun(#{'ID' := Val}) -> Val =:= ID end, Leases1)),
     timer:sleep(NewTTL * 1000 + 1000),
-    {ok, #{leases := []}} = eetcd_lease:leases(?Name),
-    
+    Leases2 = list_leases(?Name),
+    ?assertEqual([], Leases2),
+
     {error, {grpc_error, #{'grpc-status' := 5}}} = eetcd_lease:revoke(?Name, ID),
     ok.
 
 lease_keepalive_once(_Config) ->
     TTL = 2,
     {ok, #{'ID' := ID, 'TTL' := NewTTL}} = eetcd_lease:grant(?Name, TTL),
-    {ok, #{leases := [#{'ID' := ID}]}} = eetcd_lease:leases(?Name),
-    timer:sleep(1000),
+    Leases1 = list_leases(?Name),
+    [#{'ID' := ID} | _] = Leases1,
+    timer:sleep(900),
     {ok, _Pid} = eetcd_lease:keep_alive_once(?Name, ID),
     timer:sleep(NewTTL * 1000 + 200),
```

Comment:
I set this the wrong way.
It should be :
```
timer:sleep(NewTTL*1000 - 300),
```

Reply:
OK, good catch. I've updated the value per your suggestion and indeed tests pass 20/20 times now 👍